astrapy/idiomatic/database.py

```diff
@@ -91,6 +92,35 @@ def __init__(
             caller_name=caller_name,
             caller_version=caller_version,
         )
+        self.codec_options = []
+        self.read_preference = "NEAREST"
+
+        self.client_options = {
+            'token': "",
+            'api_endpoint': "",
+            'api_path': "",
+            'api_version': "",
+            'namespace': "",
+            'caller_name': "",
+            'caller_version': ""
+        }
+
+        astraDBOps = AstraDBOps(token=token)
+
+        # Get the database object and name
+        if "-" in api_endpoint:
+            self.dbid = api_endpoint.split('/')[2].split('.')[0][:36]
+            
+            details = astraDBOps.get_database(database=self.dbid)
+            self.name = details['info']['name']
+            self.region = details['info']['region']
+            self.database = {'id': self.dbid, 'name': self.name, 'region': self.region}
+
+        else:
+            self.name = ""
+            self.region = ""
+            self.database = ""
```

Comment:
Perhaps a better typing would be less surprising, i.e. `self.database: Optional[Dict[str, Any]]` or something like that.